soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+		self.otherPts.remove(closestPt)
+
+		# remove the point that's diagonal of the closest point
+		farthestPt = max(self.otherPts,
+						key=lambda pt: pt.dist_to(main.ball().pos))
+
+		self.otherPts.remove(farthestPt)
+
+
+		# move the other two robots to the other point locations
+		self.add_subbehavior(skills.move.Move(self.otherPts[0]), 'move1', required = False)
+		self.add_subbehavior(skills.move.Move(self.otherPts[1]), 'move2', required = False)
+		# send the closest robot to capture the ball
+		self.add_subbehavior(skills.capture.Capture(), 'capture')
+
+	def on_exit_setup(self):
+		self.remove_subbehavior('capture')
+
+	def on_enter_passing(self):
+		# get the point passing from
+		kickFrom = min(self.square_points,
+					   key=lambda pt: pt.dist_to(main.ball().pos))
+		# get the index of where you're passing from
+		kickFromIdx = self.square_points.index(kickFrom)
+		# decide which direction you're passing to
+		self.safer_pass()
+		# based on which direction you're passing to pick that index
+		kickToIdx = (kickFromIdx + self.direction) % len(self.square_points)
+		# get that point
+		kickToPt = self.square_points[kickToIdx]
+
+		# remove the subbehavior that's not recieving the pass
+		if (kickToPt is self.otherPts[0]) :
+			self.remove_subbehavior('move1')
+			self.text = 'move2'
+		else :
+			self.remove_subbehavior('move2')
+			self.text = 'move1'
+
+		# while pass is preparing get ready for the next pass
+		premove = skills.move.Move(self.square_points[(kickToIdx + self.direction) % len(self.square_points)])
+		# get ready to pass the ball
+		self.passing_machine = tactics.coordinated_pass.CoordinatedPass(kickToPt)
+
+		#if (self.has_subbehavior_with_name('Chasing')) :
+		#	self.remove_subbehavior('Chasing')
+
+		self.add_subbehavior(self.passing_machine, 'passer', required = True, priority = 10)
+		self.add_subbehavior(premove, 'premove', required = True)
+
+
+		# add a robot to chase after the ball.
+		if (not self.has_subbehavior_with_name('Chasing')):
+			self.chaser = skills.move.Move(self.cut_of_pos(main.ball().pos))
+			self.add_subbehavior(self.chaser, 'Chasing', required = True, priority = 2)
+
+		# if we have too many robots isolate one of the robots so they don't help in the play
+		if (not self.has_subbehavior_with_name('Iso') and len(main.our_robots()) == 6) :
+			self.iso = skills.move.Move(robocup.Point(-constants.Field.Width/2, 0))
+			self.add_subbehavior(self.iso,'Iso', required = True, priority = 1)
+			
+	# remove subbehaviors 
+	def on_exit_passing(self):
+		self.remove_subbehavior(self.text)
+		self.remove_subbehavior('passer')
+		self.remove_subbehavior('Chasing')
+		self.remove_subbehavior('premove')
+
+	# takes in the square points and form lines and create a square on the field
+	def paint_the_field(self):
+		for i in range(len(self.square_points)) :
+			main.system_state().draw_line(robocup.Line(self.square_points[i], self.square_points[(i + 1) % 4]), (135, 0, 255), "Square")
```

Comment:
There is a `draw_polygon` where you can pass in a rectangle if you want it filled in. It's a personal preference